packages/fx-core/src/component/driver/deploy/azure/azureStorageDeployDriver.ts

```diff
@@ -112,6 +116,9 @@ export class AzureStorageDeployDriverImpl extends AzureDeployImpl {
     );
     const responses = await Promise.all(tasks);
     const errorResponse = responses.find((res) => res.errorCode);
+    if (errorResponse?._response?.status === HttpStatusCode.INTERNAL_SERVER_ERROR) {
```

Comment:
I am a little curious about why this part is to throw server error and the next line is to throw user error. can they be merged? I will find you to learn more offline

Reply:
This error is thrown when the service is down or cannot respond properly. The message instructs the user to try again or to check the status of the Azure service.